pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -703,39 +687,41 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *v1.Serv
 
 		glog.V(4).Infof("Pool for listener %s: %s", listener.ID, pool.ID)
 		members, err := getMembersByPoolID(lbaas.network, pool.ID)
-		if err != nil && !isNotFound(err) {
+		if err != nil && err != ErrNotFound {
 			return nil, fmt.Errorf("Error getting pool members %s: %v", pool.ID, err)
 		}
-		for _, node := range nodes {
-			addr, err := nodeAddressForLB(node)
-			if err != nil {
-				if err == ErrNotFound {
-					// Node failure, do not create member
-					glog.Warningf("Failed to create LB pool member for node %s: %v", node.Name, err)
-					continue
-				} else {
-					return nil, fmt.Errorf("Error getting address for node %s: %v", node.Name, err)
+		if !isNotFound(err) && members != nil {
+			for _, node := range nodes {
+				addr, err := nodeAddressForLB(node)
+				if err != nil {
+					if err == ErrNoAddressFound {
```

Comment:
I think this `ErrNotFound` -> `ErrNoAddressFound` and the additional "ignore if not found" check around this foreach-node loop are the behavioural changes in this PR, and the rest is refactoring.  Is that correct or did I miss any others?